### PR TITLE
Add contributor docs / guides, add pr + issue templates

### PR DESCRIPTION
**Pull request by aaronlademann-wf**
Pretty self-explanatory...

I added `CONTRIBUTING.md`, along with a PR / ISSUE template, and an update to the main readme with a word about our friend `dartfmt`, and our versioning scheme.

__FYA:__ @jacehensley-wf @clairesarsam-wf @greglittlefield-wf @joelleibow-wf

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Raven
Number of Findings: 0

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+&nbsp;
+
+
+### Pull requests
+
+Good pull requests - patches, improvements, new features - are a fantastic help. They should remain focused in scope and avoid containing unrelated commits.
+
+__Please ask first__ before embarking on any significant pull request (e.g. implementing features, refactoring code, porting to a different language), otherwise you risk spending a lot of time working on something that the project's lead developers might not want to merge into the project.
+
+Please adhere to the [Dart Style Guide][dart-style-guide] for all changes contained in your pull requests.
```
Comment (greglittlefield-wf) [inline .github/CONTRIBUTING.md]: Broken link

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+   ```bash
+   git checkout -b <topic-branch-name>
+   ```
+
+4. Commit your changes in logical chunks. Please adhere to these [git commit message guidelines](#git-commit-message-standards) or your code is unlikely be merged into the master branch. Optionally, you can use Git's [interactive rebase](https://help.github.com/articles/interactive-rebase) feature to tidy up your commits before making them public.
+
+5. Write tests for your changes.  
+  1. There are no exceptions.  
+  2. If your PR doesn't have accompanying test coverage - your code 
+     will not get merged.
```
Comment (greglittlefield-wf) [inline .github/CONTRIBUTING.md]: This wording isn't super friendly. Not saying we shouldn't require changes to be tested, but may be good to include something like: "if you're having trouble, reach out in your PR about how to best go about testing your changes."

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+6. Locally merge the upstream master branch into your topic branch:
```
Comment (greglittlefield-wf) [inline .github/CONTRIBUTING.md]: May want to note that this is only necessary if there are merge conflicts

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+```bash
+pub serve
+```
+
+When you're ready to run the tests... in a separate terminal:
+
+```bash
+pub run test -p content-shell --pub-serve=8081 test/over_react_test.dart
+```
```
Comment (greglittlefield-wf) [inline .github/CONTRIBUTING.md]: There are the VM tests that we want to run as well, but wouldn't get run using these instructions. We should just recommend using `pub run dart_dev test`.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+&nbsp;
+&nbsp;
+
+
+
+## Developer Workflow
+
+The `over_react` developer workflow couldn't be any more simple!
+
```
Comment (greglittlefield-wf) [inline .github/CONTRIBUTING.md]: "To serve the docs/examples/whatever, run:"

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+__HOW IT WAS FIXED:__
+
+
+
+__TESTING SUGGESTIONS:__
+
+
+
+__POTENTIAL AREAS OF REGRESSION:__
```
Comment (greglittlefield-wf) [inline .github/PULL_REQUEST_TEMPLATE.md]: #supernit...

I think using markdown headers and normal casing makes this much more readable:
```markdown
## Ultimate problem:



## How it was fixed:



## Testing suggestions:



## Potential areas of regression:



---

__FYA:__ @greglittlefield-wf @aaronlademann-wf @jacehensley-wf @clairesarsam-wf @joelleibow-wf
```

> ## Ultimate problem:
>
>
>
> ## How it was fixed:
>
>
>
> ## Testing suggestions:
>
>
>
> ## Potential areas of regression:
>
>
> --- 
>
> __FYA:__ `@greglittlefield-wf` `@aaronlademann-wf` `@jacehensley-wf` `@clairesarsam-wf` `@joelleibow-wf`

File: README.md [hunk 7]
```diff
@@ -587,6 +601,231 @@ that you get for free from OverReact, you're ready to start building your own cu
...
+
+* __ALWAYS__ set a default / intial value for `props` / `state` fields, 
+and document that value in a comment.
+
+  _Why?_ Without default prop values for bool fields, they could be 
+  `null` - which is extremely confusing and can lead to a lot of 
+  unnecessary null-checking in your business logic. 
+
+  Also, if you fail to provide an initial value for all the fields in 
+  `state`, your component will fail to build at run-time.
```
Comment (greglittlefield-wf) [inline README.md]: This isn't really accurate—or maybe the wording is just off?

File: README.md [hunk 8]
```diff
@@ -587,6 +601,231 @@ that you get for free from OverReact, you're ready to start building your own cu
...
+
+    @State()
+    DropdownButtonState extends UiState {
+      bool isOpen;
+    }
+    ```
+
+&nbsp;
+
+* __AVOID__ specifying more than one cascading prop setter on the same line.
```
Comment (greglittlefield-wf) [inline README.md]: This should be grouped with the first couple guidelines.

Also, would it make sense to have separate sections for guidelines around consuming components and guidelines around building components?

File: README.md [hunk 9]
```diff
@@ -19,7 +19,9 @@
 * __[DOM components and props](#dom-components-and-props)__
 * __[Building custom components](#building-custom-components)__
     * __[Component Boilerplates](#component-boilerplate-templates)__
+    * __[Component Formatting](#component-formatting)__
```
Comment (greglittlefield-wf) [inline README.md]: See my penultimate comment; may be good to have a "component formatting" set of guidelines in the first level of this list, and then "component best practices" under this "Building custom components" section.

**Comment by aaronlademann-wf** [comment]
@greglittlefield-wf all feedback addressed.

**Comment by greglittlefield-wf** [comment]
+1

**Comment by aaronlademann-wf** [comment]
@leviwith-wf this is ready for merge.

**Comment by leviwith-wf** [comment]
- Docs look good, read well

QA +10 

Merging.

**Comment by colefeisthamel-wf** [comment]
RM +1 manually reviewed dependencies

**Pull request merged**